src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7379,5 +7379,21 @@ internal string GetDebuggerDisplay()
                 return pooledBuilder.ToStringAndFree();
             }
         }
+
+        private sealed class NullabilityInfoTypeComparer : IEqualityComparer<(NullabilityInfo info, TypeSymbol type)>
```

Comment:
>NullabilityInfoTypeComparer [](start = 29, length = 27)

Do we need this class? It looks like the uses don't need to pass comparers.

Reply:
Yes, we need to be able to compare the full value of the types when adding them for the semantic model. We get incorrect type info out if we don't